DependencyInjection/Compiler/StaticAsseticHelperPass.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Symfony\Bundle\AsseticBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+class StaticAsseticHelperPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('assetic.helper.static')) {
+            return;
+        }
+
+        $definition = $container->getDefinition('assetic.helper.static');
+
+        if (!method_exists($definition, 'setShared') && 'request' === $container->getDefinition('templating.helper.assets')->getScope()) {
```

Comment:
please also add a check whether ``templating.helper.assets`` is defined to avoid issue if the PHP engine is not enabled